src/nvim/edit.c

```diff
@@ -5280,14 +5281,15 @@ insertchar (
 
       (*mb_char2bytes)(c, buf);
       buf[cc] = NUL;
-      ins_char_bytes(buf, cc);
+      ins_char_bytes(buf, cc, false);
       AppendCharToRedobuff(c);
     } else {
-      ins_char(c);
-      if (flags & INSCHAR_CTRLV)
+      ins_char(c, true);
```

Comment:
This is what makes marks move when chars are inserted. The test for this is marked pending as it always fails. Typing in neovim will work if i type at a slow speed. If i mash heaps of keys the mark will be deleted :/